### PR TITLE
22 plane primitive

### PR DESCRIPTION
**Pull request by maelemiel**
This pull request introduces a new `Plane` primitive to the Raytracer project, implementing its functionality, integration, and unit tests. The key changes include the implementation of the `Plane` class, its header file, integration into the build system, and comprehensive unit tests.

### Implementation of the Plane Primitive:
* [`src/scene/primitives/Plane.cpp`](diffhunk://#diff-8ec16250329587e176b6516d5c68c3db87bb7c8f0aa9895f81bdb7d23e163d58R1-R141): Added the implementation of the `Plane` class, which represents an infinite plane primitive. This includes methods for intersection testing, setting and retrieving transformations and colors, calculating normals, and cloning the object.
* [`src/scene/primitives/Plane.hpp`](diffhunk://#diff-f9bed98521ea28ba9a28073adafe1c0ea8958e611df2ed52b5ecdbdbc584f329R1-R105): Added the header file for the `Plane` class, defining its interface and private members, including axis alignment, position, color, and transformation matrices.

### Integration into the Build System:
* [`tests/CMakeLists.txt`](diffhunk://#diff-4461c617ceae0c7e0206622aacdf555aedd62eb129c2efb74c84fa1567bcbe0dR9): Added `test_Plane.cpp` to the list of test sources, ensuring the Plane tests are included in the build process.

### Unit Tests for the Plane Class:
* [`tests/test_Plane.cpp`](diffhunk://#diff-32744d5b13ec031004dd8d3b068ffb42ea70a36c13e62fae6cc01888e75735d8R1-R306): Added extensive unit tests for the `Plane` class, covering construction, intersection behavior (including edge cases like parallel rays and transformed planes), color and transformation setters/getters, and the cloning method. Helper functions and macros were also added to facilitate detailed vector and intersection comparisons.

### CODE REVIEW

**Review by SantiagoPidji — DISMISSED** [review]

File: src/scene/primitives/Plane.hpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+   * @return A new instance of this plane
+   */
+  std::shared_ptr<IPrimitive> clone() const override;
+
+ private:
+  Vector3D _point;              ///< A point on the plane
+  Vector3D _normal;             ///< The normal vector of the plane
+  Color _color;                 ///< The color of the plane
+  Transform _transform;         ///< Transformation applied to the plane
+  Transform _inverseTransform;  ///< Inverse transformation
```
[the remark this answers is not in the document]
Comment (maelemiel) [reply src/scene/primitives/Plane.hpp]: fait !

**Review by Arkteus — APPROVED** [review]
:+1: 
:smile:
